homeassistant/components/microbees/config_flow.py

```diff
@@ -0,0 +1,87 @@
+"""Config flow for microBees integration."""
+from collections.abc import Mapping
+import logging
+from typing import Any
+
+from homeassistant import config_entries
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import (
+    aiohttp_client,
+    config_entry_flow as cv,
+    config_entry_oauth2_flow,
+)
+
+from .api import get_api_scopes
+from .const import DOMAIN, VERSION
+from .microbees import MicroBeesConnector
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConfigFlow(config_entry_oauth2_flow.AbstractOAuth2FlowHandler, domain=DOMAIN):
+    """Handle a config flow for microBees."""
+
+    CONFIG_SCHEMA = cv.config_entry_only_config_schema
+    DOMAIN = DOMAIN
+    VERSION = VERSION
+
+    @property
+    def logger(self) -> logging.Logger:
+        """Return logger."""
+        return logging.getLogger(__name__)
+
+    @property
+    def extra_authorize_data(self) -> dict:
+        """Extra data that needs to be appended to the authorize url."""
+        scopes = get_api_scopes(self.flow_impl.domain)
+        return {"scope": " ".join(scopes)}
+
+    async def async_oauth_create_entry(self, data: dict) -> FlowResult:
+        """Create an oauth config entry or update existing entry for reauth."""
+        existing_entry = await self.async_set_unique_id(DOMAIN)
+        if existing_entry:
+            self.hass.config_entries.async_update_entry(existing_entry, data=data)
+            await self.hass.config_entries.async_reload(existing_entry.entry_id)
+            return self.async_abort(reason="reauth_successful")
+
+        microBees = MicroBeesConnector(
+            session=aiohttp_client.async_get_clientsession(self.hass),
+            token=data["token"]["access_token"],
+        )
+
+        current_user = await microBees.getMyProfile()
+
+        data["id"] = current_user.id
+
+        name = current_user.firstName + " " + current_user.lastName
+        data["name"] = name
+
+        await self.async_set_unique_id(current_user.id)
```

Comment:
Okay so when looking at the config flow and the init, how many entries do you allow? From the first line I see you only allow one (as the unique_id is set to a static value) but later on in the config flow we set it to the current_user.id.

Reply:
Question is still open